src/main/kotlin/com/terraformation/backend/nursery/api/BatchesController.kt

```diff
@@ -37,6 +45,24 @@ class BatchesController(
     val insertedRow = batchStore.create(payload.toRow())
     return BatchResponsePayload(BatchPayload(insertedRow))
   }
+
+  @ApiResponse404
+  @ApiResponse412
```

Comment:
Am I correct in understanding that these annotation are meant to catch exceptions and turn them into error responses? One thing that's not connecting for me is how they know which exception to catch for each of the 4xx responses.

Reply:
Not quite; these annotations are for the benefit of the OpenAPI schema generator. With these annotations, the generated API schema will list HTTP 404 and 412 as possible response codes for this endpoint. The annotations don't have any effect on the actual exception handling.

However, you're asking the right question. There _is_ a place where exceptions get mapped to HTTP responses: https://github.com/terraware/terraware-server/blob/main/src/main/kotlin/com/terraformation/backend/api/ControllerExceptionHandler.kt

Reply:
Awesome, thanks for the explanation and link!